hazelcast/src/main/java/com/hazelcast/internal/networking/nonblocking/NonBlockingIOThread.java

```diff
@@ -245,7 +254,7 @@ private void coolDown() {
     }
 
     private void selectLoop() throws IOException {
-        while (!isInterrupted()) {
+        while (!stop) {
```

Comment:
Noting else than shutdown should interrupt a thread. Interrupt is not a general purpose method everybody can invoke and see what happens.